From d3fd230a9b5fd43c8013931fd1f6167671836251 Mon Sep 17 00:00:00 2001
From: Omri Itach <omrii@marvell.com>
Date: Wed, 29 Jan 2014 15:59:40 +0200
Subject: [PATCH 1320/1825] fix: a375: disable external SMI support

https://github.com/MISL-EBU-System-SW/misl-windriver.git linux-3.4.69-14t2-read
commit a3bda64c74b9b9ef5ef2d26ca1bb261fead04b2c

	- Since not using any external RGMII PHY, external SMI control is not needed
	- updated A375 MPP's to disable CPU and Switch SMI control
	- removed unused SmiCtrl routines and defines for A375 and A38x family

Change-Id: I19491a6e4050f112e0f8f6ab1247ad2e5e356678
Signed-off-by: Omri Itach <omrii@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/5372
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Signed-off-by: Zhong Hongbo <hongbo.zhong@windriver.com>
---
 .../armada_38x_family/ctrlEnv/mvCtrlEnvLib.h       |    7 -------
 1 files changed, 0 insertions(+), 7 deletions(-)

diff --git a/arch/arm/mach-armada38x/armada_38x_family/ctrlEnv/mvCtrlEnvLib.h b/arch/arm/mach-armada38x/armada_38x_family/ctrlEnv/mvCtrlEnvLib.h
index d6f7414..edb9312 100644
--- a/arch/arm/mach-armada38x/armada_38x_family/ctrlEnv/mvCtrlEnvLib.h
+++ b/arch/arm/mach-armada38x/armada_38x_family/ctrlEnv/mvCtrlEnvLib.h
@@ -74,12 +74,6 @@
 #include "ctrlEnv/mvCtrlEnvAddrDec.h"
 
 
-/* This enumerator describes the possible SMI control options */
-typedef enum _mvSMIctrl {
-	CPU_SMI_CTRL,
-	SWITCH_SMI_CTRL,
-} MV_SMI_CTRL;
-
 /* typedefs */
 typedef MV_STATUS (*MV_WIN_GET_FUNC_PTR)(MV_U32, MV_U32, MV_UNIT_WIN_INFO*);
 
@@ -251,7 +245,6 @@ MV_U32    ctrlSizeRegRoundUp(MV_U32 size, MV_U32 alignment);
 MV_U32 mvCtrlSysRstLengthCounterGet(MV_VOID);
 MV_STATUS ctrlWinOverlapTest(MV_ADDR_WIN *pAddrWin1, MV_ADDR_WIN *pAddrWin2);
 MV_STATUS ctrlWinWithinWinTest(MV_ADDR_WIN *pAddrWin1, MV_ADDR_WIN *pAddrWin2);
-MV_VOID mvCtrlSmiMasterSet(MV_SMI_CTRL smiCtrl);
 MV_VOID mvCtrlPwrClckSet(MV_UNIT_ID unitId, MV_U32 index, MV_BOOL enable);
 MV_BOOL mvCtrlPwrClckGet(MV_UNIT_ID unitId, MV_U32 index);
 MV_BOOL mvCtrlIsBootFromNOR(MV_VOID);
-- 
1.7.5.4

